### PR TITLE
JDK-8273387: remove some unreferenced gtk-related functions

### PR DESCRIPTION
**Pull request by MBaesken**
Please review this small change.
Looks like there are a few functions, like gtk2_paint_diamond, that are unreferenced and can be removed.

Thanks, Matthias

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273387](https://bugs.openjdk.java.net/browse/JDK-8273387): remove some unreferenced gtk-related functions

### Reviewers
 * [Pankaj Bansal](https://openjdk.java.net/census#pbansal) (@pankaj-bansal - **Reviewer**)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5384/head:pull/5384` \
`$ git checkout pull/5384`

Update a local copy of the PR: \
`$ git checkout pull/5384` \
`$ git pull https://git.openjdk.java.net/jdk pull/5384/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5384`

View PR using the GUI difftool: \
`$ git pr show -t 5384`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5384.diff">https://git.openjdk.java.net/jdk/pull/5384.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/unix/native/libawt_xawt/awt/gtk2_interface.c [hunk 1]
```diff
@@ -1850,19 +1850,6 @@ static void gtk2_paint_check(WidgetType widget_type, gint synth_state,
             x, y, width, height);
 }
 
-static void gtk2_paint_diamond(WidgetType widget_type, GtkStateType state_type,
```
Comment (pankaj-bansal) [inline src/java.desktop/unix/native/libawt_xawt/awt/gtk2_interface.c:1853]: I think the function pointer initialiasation of this function `fp_gtk_paint_diamond` should also be removed along with removing the function definition.

Comment (MBaesken) [reply src/java.desktop/unix/native/libawt_xawt/awt/gtk2_interface.c:1853]: Thanks for the hint , I removed the initialization you mentioned.

Best regards, Matthias

Comment (pankaj-bansal) [reply src/java.desktop/unix/native/libawt_xawt/awt/gtk2_interface.c:1853]: I think the declaration of function pointer should also be removed. So following should be removed from line no 150
static void (*fp_gtk_paint_diamond)(GtkStyle* style, GdkWindow* window,
        GtkStateType state_type, GtkShadowType shadow_type,
        GdkRectangle* area, GtkWidget* widget, const gchar* detail,
        gint x, gint y, gint width, gint height);

Comment (MBaesken) [reply src/java.desktop/unix/native/libawt_xawt/awt/gtk2_interface.c:1853]: Thanks, I removed the declaration you mentioned.
Best regards, Matthias

**Review by pankaj-bansal — APPROVED** [review]
Looks good to me now. I have run SwingSet2 with both gtk2 and gtk3

**Review by RealCLanger — APPROVED** [review]

**Pull request closed without merging**